chia/rpc/wallet_rpc_api.py

```diff
@@ -980,6 +1007,78 @@ async def send_transaction_multi(self, request) -> EndpointResult:
         # Transaction may not have been included in the mempool yet. Use get_transaction to check.
         return {"transaction": transaction, "transaction_id": tr.name}
 
+    async def get_coins_by_type(self, request) -> EndpointResult:
+        wallet_id = int(request["wallet_id"])
+        coin_type = CoinType(request["coin_type"])
+        start = request.get("start", 0)
+        end = request.get("end", 50)
+        reverse = request.get("reverse", False)
+
+        coins = await self.service.wallet_state_manager.coin_store.get_coin_records_between(
+            wallet_id,
+            start,
+            end,
+            reverse=reverse,
+            coin_type=coin_type,
+        )
+        return {
+            "coins": [
+                {
+                    "coin_id": coin.name().hex(),
+                    "coin_type": coin.coin_type,
+                    "amount": coin.coin.amount,
+                    "puzzle_hash": coin.coin.puzzle_hash.hex(),
+                    "parent_coin": coin.coin.parent_coin_info.hex(),
+                    "metadata": self.service.wallet_state_manager.deserialize_coin_metadata(
+                        coin.metadata.blob, coin_type, True
+                    )
+                    if coin.metadata is not None
+                    else None,
+                    "confirmed_height": coin.confirmed_block_height,
+                    "spent_height": coin.spent_block_height,
+                }
+                for coin in coins
+            ],
+            "wallet_id": wallet_id,
+        }
+
+    async def spend_clawback_coins(self, request) -> EndpointResult:
```

Comment:
It's a good suggestion, but I think we can do this in another PR. I did a dry run and found 170+ errors that need to fix.

Reply:
I think it makes sense to do that in another PR, but that PR should land before this one, and this should be rebased on top of it. Otherwise you risk making it worse (like in this instance)

Reply:
Just want to confirm, do you mean we should create different request classes for different APIs? That's huge work. Or you mean add like Dict[str, Any] for all requests parameters?

Reply:
The most important objective is to have type safety. I don't have particularly strong opinions on how to get there (not yet at least). How about `Dict[str, str]`? If this is just parsed from JSON, we could use something like:

```
Json = Union[Dict[str, Json], List[Json], int, str]
```